src/classes/RD_RecurringDonations.cls

```diff
@@ -884,29 +1169,65 @@ public class RD_RecurringDonations {
         
         for (Integer i=0; i<recurringDonationsToEvaluate.size(); i++) {
             npe03__Recurring_Donation__c rd = recurringDonationsToEvaluate[i];
-            
+
             //don't try to evaluate orphaned recurring donations
             if (rd.npe03__Contact__c == null && rd.npe03__Organization__c == null) {
                 recurringDonationsToEvaluate.remove(i);
                 //decrement counter after removing value to avoid going over the end of the array
                 i--;
                 continue;
             }
-
             if (rd.npe03__Open_Ended_Status__c != openlabel && rd.npe03__Open_Ended_Status__c != closedlabel) {
                 listRDNonOpen.add(rd);
             } else {
                 listRDOpen.add(rd);
             }
         }
 
+        TDTM_Runnable.DmlWrapper dmlWrapper = new TDTM_Runnable.DmlWrapper();
+
         if (listRDNonOpen.size() > 0) {
-            //clear existing open opps and recreate
-            delete [select id from Opportunity where npe03__Recurring_Donation__c IN :listRDNonOpen and isClosed = false];
-            insertOppsOnRecurringDonationInsert(listRDNonOpen);
+
+            List<Opportunity> opps = [SELECT Id
```

Comment:
Same comment here about combining these two queries into a single one and then split the data into 2 collections to pass into their respective handlers.